camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/BrooklynDslCommon.java

```diff
@@ -327,25 +337,74 @@ public FormatString(String pattern, Object[] args) {
 
         private static final long serialVersionUID = 8878388748085419L;
 
-        private String typeName;
-        private Class<?> type;
-        private Map<String,Object> fields, config;
+        private final String typeName;
+        private final String factoryMethodName;
+        private final Class<?> type;
+        private final List<Object> constructorArgs, factoryMethodArgs;
+        private final Map<String, Object> fields, config;
+
+        public DslObject(
+                String typeName,
+                List<Object> constructorArgs,
+                Map<String, Object> fields,
+                Map<String, Object> config) {
```

Comment:
I'd use `Map<String, ?>` if we can get away with it - makes it easier for the caller.
But this is pretty internal stuff, so I don't really care.